main.py

```diff
@@ -184,11 +251,24 @@ def get_name_tag(resource_type, name):
 
 private_sg.delete()
 public_sg.delete()
+elbv2.delete_listener(ListenerArn=listener_arn)
+elbv2.delete_target_group(TargetGroupArn=target_group_arn)
+elbv2.delete_load_balancer(LoadBalancerArn=alb_arn)
+elbv2.get_waiter('load_balancers_deleted').wait(LoadBalancerArns=[alb_arn])
+
+while True:
+    try:
+        alb_sg.delete()
+        break
+    except ClientError:
+        print(f'Waiting for the dependencies of {alb_sg} to be cleaned up.')
+        sleep(5)
+
```

Comment:
I came up with not so elegant but working solution, to make sure that all the dependencies of the ALB-sg is cleaned up.